### PR TITLE
Fixes:#2837 Updated triggers for IsItUp

### PR DESCRIPTION
**Pull request by gautamkrishnar**
Fixes #2954 
Added triggers and regexp to check the triggers....

@moollaza Is this regexp okay?.  I need your help on regexp...

---

Instant Answer Page: http://duck.co/ia/view/is_it_up

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @rasikapohankar

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Spice/IsItUp.pm [hunk 1]
```diff
@@ -8,14 +8,17 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
 use Net::Domain::TLD qw(tld_exists);
 use Data::Validate::IP qw(is_ipv4);
 
-triggers query_lc => qr/^((?:is\s|))(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)\s(?:up|down|working|online|status)\?*$/i;
+
+triggers start => "isitup", "isitdown" , "is";
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: we can remove `is`

Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: we can probably add `status of` here

File: lib/DDG/Spice/IsItUp.pm [hunk 2]
```diff
@@ -8,14 +8,17 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
...
+triggers end => "up", "down", "working", "online", "status";
+
 
 spice to => 'https://isitup.org/$1.json?callback={{callback}}';
 
 spice proxy_cache_valid => "418 1d";
 
 handle matches => sub {
-
+    return unless grep {$req->query_lc}=~ /^((?:is\s|isitup\s|isitdown\s|))(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)\s(?:up|down|working|online|status)\?*$/i;
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: @gautamkrishnar no need for `grep`, you can use `m/$regex/`

You also need to change the `handle` to `remainder_lc` or `query_lc`. Depending which you choose you'll have to adjust this regex.

File: lib/DDG/Spice/IsItUp.pm [hunk 3]
```diff
@@ -8,14 +8,16 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
...
+triggers start => "isitup", "isitdown" , "status of";
+triggers end => "up", "down", "working", "online", "status";
 
 spice to => 'https://isitup.org/$1.json?callback={{callback}}';
 
 spice proxy_cache_valid => "418 1d";
 
 handle matches => sub {
-
+    return unless {$req->query_lc}=~ m/^((?:isitup\s|isitdown\s|status\sof|))(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)\s(?:up|down|working|online|status)\?*/;
```
Comment (gautamkrishnar) [inline lib/DDG/Spice/IsItUp.pm]: @moollaza shall i have to add `*$/i` at the end of regexp?

Comment (gautamkrishnar) [inline lib/DDG/Spice/IsItUp.pm]: Is this regexp okay...

File: lib/DDG/Spice/IsItUp.pm (hunk 2 above)
Comment (rasikapohankar) [inline lib/DDG/Spice/IsItUp.pm]: @gautamkrishnar @moollaza We will need to use `handle query_lc` as it is checking if the first group matched in the regex (`is` according to the earlier regex, `isitup`, `isitdown`,... now) on Line 42 here. And also change `$_[0]`, `$_[1]`... to `$1`, `$2`... for the matches in the handle function

@moollaza Why do we need to use `m` in the regular expression?

File: lib/DDG/Spice/IsItUp.pm [hunk 4]
```diff
@@ -8,14 +8,16 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
 use Net::Domain::TLD qw(tld_exists);
 use Data::Validate::IP qw(is_ipv4);
 
-triggers query_lc => qr/^((?:is\s|))(?:https?:\/\/)?([\p{Alnum}\-]+(?:\.[\p{Alnum}\-]+)*?)(?:(\.\pL{2,})|)\s(?:up|down|working|online|status)\?*$/i;
+
+triggers start => "isitup", "isitdown" , "status of";
+triggers end => "up", "down", "working", "online", "status";
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: We can add `up right now` and `down right now` for "is twitter up right now"

File: lib/DDG/Spice/IsItUp.pm (hunk 3 above)
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: @gautamkrishnar looks like this no longer handles the format `is $url (up|down|working|...)`

File: lib/DDG/Spice/IsItUp.pm [hunk 5]
```diff
@@ -8,14 +8,16 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
...
 
 spice to => 'https://isitup.org/$1.json?callback={{callback}}';
 
 spice proxy_cache_valid => "418 1d";
 
 handle matches => sub {
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: @gautamkrishnar you need to replace `handle matches` as that is designed to work with a `regex` trigger.

**Comment by moollaza** [comment]
@gautamkrishnar once the tests are passing I can deploy this to Beta. The tests will also ensure all the same queries work that we expect. Please add more cases to the tests for any new triggers we've added as well 👍

File: lib/DDG/Spice/IsItUp.pm (hunk 3 above)
Comment (gautamkrishnar) [inline lib/DDG/Spice/IsItUp.pm]: okay. i will fix that

**Comment by gautamkrishnar** [comment]
@moollaza  it is now working for `is www.apple.com down` but not working for `isitup www.apple.com...
please help me on this...
is there a way to keep the `triggers end optional. So that it will match when we give the above query...

**Comment by rasikapohankar** [comment]
Hi @gautamkrishnar, Are you still working on this one?

**Comment by gautamkrishnar** [comment]
@rasikapohankar  Ops!  I had totally forgotten this one.  Thanks for notifying me.  I will fix this one soon.. Since I have my exams going on I couldn't work on it right now.  I will update the PR this weekend.

**Comment by gautamkrishnar** [comment]
@rasikapohankar i can't work on this one due to my academic schedules... Can you please take this one?
cc: @GuiltyDolphin @moollaza

**Comment by rasikapohankar** [comment]
@gautamkrishnar thanks, I'll finish it. :+1:

**Pull request closed without merging**